addons/mail/static/src/discuss/call/common/discuss_app_model_patch.js

```diff
@@ -5,23 +5,20 @@ import { Record } from "@mail/core/common/record";
 
 import { patch } from "@web/core/utils/patch";
 
-patch(DiscussApp, {
-    new(data) {
-        const app = super.new(data);
-        Record.onChange(app, "ringingThreads", () => {
-            if (app.ringingThreads.length > 0) {
-                this.env.services["mail.sound_effects"].play("incoming-call", { loop: true });
-            } else {
-                this.env.services["mail.sound_effects"].stop("incoming-call");
-            }
-        });
-        return app;
-    },
-});
-
 patch(DiscussApp.prototype, {
     setup() {
         super.setup();
-        this.ringingThreads = Record.many("Thread");
+        this.ringingThreads = Record.many("Thread", {
+            /** @this {import("models").DiscussApp} */
+            onUpdate() {
+                if (this.ringingThreads.length > 0) {
+                    this._store.env.services["mail.sound_effects"].play("incoming-call", {
+                        loop: true,
+                    });
```

Comment:
no risk to play twice? if adding more ringing

Reply:
onUpdate is called only once at end of update cycle. It's behaviourally equivalent to `onChange()` in diff before